overlord/snapstate/autorefresh_gating_test.go

```diff
@@ -1950,7 +1950,6 @@ func (s *snapmgrTestSuite) testAutoRefreshPhase2(c *C, beforePhase1 func(), gate
 		chg.AddAll(ts)
 	}
 
-	defer s.se.Stop()
```

Comment:
why these changes?

Reply:
As discussed with @Meulengracht `Stop()` in managers must be called with state unlocked, since `SnapManager` didn't implement `Stop()` before this PR, it was not caught earlier and the tests were able to get away with doing that. To fix this I am calling `Stop()` on test cleanup https://github.com/snapcore/snapd/commit/1e6dec8ec6e77e84be453d85c9e05dbc308965d7, this commit can be extracted into a separate PR, but since it was blocking my tests I added it here.